string-format rule not working on method called when obj is inside an struct.

**Describe the bug**
The string-format rule search only for function calls and methods, but if the object is
inside a structure, the method call is not found by the rule.
This is a typical pattern when we want a generic Logger interface and different implementations. In this case `Logger` is a port and `MyLogger` would be an adapter from a project with hexagonal architecture. This implementation is injected in main to a service.

**To Reproduce**
Steps to reproduce the behavior:
1. Create go file with the following code:
```golang
package main

import "fmt"

type Logger interface {
	Error(msg string, keysAndValues ...any)
}
type MyLogger struct {
}

func (m *MyLogger) Error(msg string, _ ...any) {
	fmt.Println(msg)
}

type Service struct {
	logger Logger
}

func main() {
	s := Service{
		logger: &MyLogger{},
	}
	s.logger.Error("Test")
}
```
2. I run it with the following flags & configuration file:

```shell
revive main.go
```

_config_
```toml
[rule.string-format]
  arguments = [
    ["logger.Error", "/^([^A-Z]|$)/",  "must not start with a capital letter"]]
```

**Expected behavior**
I spect it to fail with 

`./tests/main.go:23:17: must not start with a capital letter`

as the logger.Error call start with a capital letter

**Desktop (please complete the following information):**
 - OS: Gentoo
 - Version of Go: 1.20

**Additional context**
MR is created with working solution (I don't know if the best)

--- a/rule/string-format.go
+++ b/rule/string-format.go
@@ -185,11 +185,13 @@ func (w lintStringFormatRule) Visit(node ast.Node) ast.Visitor {
 	if !ok {
 		return w
 	}
+
 	// Get the name of the call expression to check against rule scope
 	callName, ok := w.getCallName(call)
 	if !ok {
 		return w
 	}
+
 	for _, rule := range w.rules {
 		if rule.scope.funcName == callName {
 			rule.Apply(call)
@@ -217,7 +219,6 @@ func (lintStringFormatRule) getCallName(call *ast.CallExpr) (callName string, ok
 		if ok {
 			return recv.Sel.Name + "." + selector.Sel.Name, true
 		}
-		return "", false
 	}
 
 	return "", false

--- a/rule/string-format.go
+++ b/rule/string-format.go
@@ -185,13 +185,11 @@ func (w lintStringFormatRule) Visit(node ast.Node) ast.Visitor {
 	if !ok {
 		return w
 	}
-
 	// Get the name of the call expression to check against rule scope
 	callName, ok := w.getCallName(call)
 	if !ok {
 		return w
 	}
-
 	for _, rule := range w.rules {
 		if rule.scope.funcName == callName {
 			rule.Apply(call)
@@ -211,10 +209,15 @@ func (lintStringFormatRule) getCallName(call *ast.CallExpr) (callName string, ok
 	if selector, ok := call.Fun.(*ast.SelectorExpr); ok {
 		// Scoped function call
 		scope, ok := selector.X.(*ast.Ident)
-		if !ok {
-			return "", false
+		if ok {
+			return scope.Name + "." + selector.Sel.Name, true
 		}
-		return scope.Name + "." + selector.Sel.Name, true
+		// Scoped function call inside structure
+		recv, ok := selector.X.(*ast.SelectorExpr)
+		if ok {
+			return recv.Sel.Name + "." + selector.Sel.Name, true
+		}
+		return "", false
 	}
 
 	return "", false

--- a/test/string-format_test.go
+++ b/test/string-format_test.go
@@ -21,7 +21,11 @@ func TestStringFormat(t *testing.T) {
 			[]interface{}{
 				"s.Method3[2]",
 				"!/^[Tt][Hh]/",
-				"must not start with 'th'"}}})
+				"must not start with 'th'"},
+			[]interface{}{
+				"s.Method4", // same as before, but called from a struct
+				"!/^[Ot][Tt]/",
+				"must not start with 'ot'"}}})
 }
 
 func TestStringFormatArgumentParsing(t *testing.T) {

--- a/testdata/string-format.go
+++ b/testdata/string-format.go
@@ -18,6 +18,16 @@ func (s stringFormatMethods) Method3(a, b, c string) {
 
 }
 
+type stringFormatMethodsInjected struct{}
+
+func (s stringFormatMethodsInjected) Method4(a, b, c string) {
+
+}
+
+type container struct {
+	s stringFormatMethodsInjected
+}
+
 func stringFormat() {
 	stringFormatMethod1("This string is fine", "")
 	stringFormatMethod1("this string is not capitalized", "") // MATCH /must start with a capital letter/
@@ -27,4 +37,9 @@ func stringFormat() {
 		d: "This string is capitalized, but ends with a period."}) // MATCH /string literal doesn't match user defined regex /[^\.]$//
 	s := stringFormatMethods{}
 	s.Method3("", "", "This string starts with th") // MATCH /must not start with 'th'/
+
+	c := container{
+		s: stringFormatMethods{},
+	}
+	c.s.Method4("Other string starts with ot") // MATCH /must not start with 'ot'/
 }

